lib/spec/staticcacher.sw-spec.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+importScripts('/base/lib/staticcacher.js');
+
+describe('Static cacher', function() {
+
+  var defaultCacheName = 'offline';
+  var defaultUrls = ['http://www.example.com/', 'http://www.mozilla.org/'];
+  var defaultStaticCacher = new StaticCacher(defaultUrls);
+
+  sinon.stub(self, 'fetch', function() {
+    var response = new Response('content', {});
+    return Promise.resolve(response);
+  });
+
+  afterEach(function() {
+    self.fetch.reset();
+    return clearDefaultCache();
+  });
+
+  function clearDefaultCache() {
+    return caches.open(defaultCacheName).then(function(cache) {
+      return cache.keys().then(function(response) {
+        response.forEach(function(element) {
+          cache.delete(element);
+        });
+      });
+    });
+  }
+
+
+  describe('The constructor', function() {
+
+    it('should accept an array of files', function() {
+      var fileList = ['file1', 'file2'];
+      var cacher = new StaticCacher(fileList);
+      expect(cacher.files).to.deep.equal(fileList);
+    });
+
+    it('should throw an error if no argument is provided', function() {
+      expect(function() {
+        new StaticCacher();
+      }).to.throw(Error, 'Invalid file list');
+    });
+
+    it('should throw an error if the argument is not an array', function() {
+      expect(function() {
+        new StaticCacher('invalidString');
+      }).to.throw(Error, 'Invalid file list');
+    });
+
+    it('should throw if the array is empty', function() {
+      expect(function() {
+        new StaticCacher([]);
+      }).to.throw(Error, 'Invalid file list');
+    });
+  });
+
+  describe('onInstall()', function() {
+
+    it('should return a promise', function() {
+      expect(defaultStaticCacher.onInstall()).to.be.an.instanceOf(Promise);
+    });
+
+    it('should use the default cache', function() {
```

Comment:
Actually not. `onInstall()` should get the job done which is to store the responses for the requests into the default cache `'offline'`. Nothing more, nothing less.

So I would assert on number of keys in `offline` and over the responses there.

Does it make sense to you?